Initial migration commit

maljac opened the pull request:
Migration of existing work from intercompany connector to odoo2odoo base connector. 

Tasks: 
- [X] Switch to odoo2odoo related naming convention instead "intercompany"
- [ ] Migrate existing unittests

Commit:
Initial migration commit

Commit:
Fixed Readme and added missing oca_dependencies

Commit:
Migration and refactoring process pending (1)

Commit:
Migration and refactoring, process pending (2)

elicoidal commented on README.md:
```diff
@@ -1,2 +1,10 @@
+Odoo2Odoo Connector Modules
+============================
+
+*Odoo2Odoo Connector enables a bi-directional communication between two (or more), sepearte Odoo instances.*
```
s/sepearte/separate

elicoidal commented on odooconnector_base/events.py:
```diff
@@ -0,0 +1,142 @@
+    if session.context.get('connector_no_export'):
+        return
+
+    _logger.debug('Record create triggered for "%s(%s)"',
```
s/Record create triggered for/Record creation triggered for

elicoidal commented on odooconnector_base/models/odooconnector.py:
```diff
@@ -0,0 +1,174 @@
+
+    exported_record = fields.Boolean(
+        string='Exported',
+        help='Indicator if the record is exported by this system or not.',
```
s/if/whether

elicoidal commented on odooconnector_base/models/product_supplierinfo.py:
```diff
@@ -0,0 +1,149 @@
+# -*- coding: utf-8 -*-
```
I'd rather have this model and purchase in a separate module odooconnector_purchase (same for sales etc.) and I would like to follow Odoo common structure
In general some users will have different needs:
- simple product or partner (use this one)
- PO2SO (use odooconnector_base+odooconnector_purchase+odooconnector_sale)
- SO2SO (use odooconnector_base+odooconnector_sale)
- etc.
  depending on the final needs we can assemble the basic models accordingly

maljac replied:
First I had the same thought about supplierinfo. But actually you might need it, even if you are not using any specific scenario (e.g. migration between databases).
Since `product.supplierinfo` is in `addons/product` I think that there is no problem in keeping it in `_base` and considering it at _basic master data_.

elicoidal replied:
actually |product.supplierinfo is defined in product.product so it makes 
sense.
I would segregate though for sales and purchases|

maljac replied:
Yes! Segregation should be definitely done.

Commit:
Migration and refactoring, process pending (3)

- Add purchase related technical module
- Add po2so scenario

Commit:
Migration and refactoring, process pending (4)

- Fix filename inconsistences
- Add po2so feature

pedrobaeza commented:
Dependency on runbot added and a rebuild launched.

Commit:
Migration and refactoring, process pending (5)

- Adds basic security groups / model access
- Adds basic ImportBinder
- Adds some methods / class documentation
- Restructured backend view
- Cleaned up dependencides in _base to purchase
- Adds basic units in _purchase

Commit:
Merge branch '8.0' into initial_base_connector

Conflicts:
	README.md

Commit:
Adds oerplib dependency to .travis.yml / Restructured backend adapter code

Commit:
Fix pep8 /pylint issues // Adds missings 'active' field in backend model.

Commit:
Fix pylint issues (2)

maljac commented:
@pedrobaeza : Could you please add `oerplib` as PIP dependency to RunBot ?

pedrobaeza commented:
No, I don't have SSH access to runbot machine. @gurneyalex, can you do it?

Commit:
Adds missing uom_po_id to product export mapping

maljac commented:
@gurneyalex Could you please add `oerplib` to runbot?

maljac commented:
@gurneyalex Is it possible to install `oerplib` on runbot?

pedrobaeza commented:
Hi, @maljac, dependency is now fixed on runbot and I have restarted runbot build.

maljac commented:
@pedrobaeza, thank you!

gurneyalex commented:
:+1:

elicoidal commented on odooconnector_base/README.rst:
```diff
@@ -0,0 +1,70 @@
+Usage
+=====
+
+.
```
Will require documentation and screenshot. Let me know if I can help here

elicoidal commented on odooconnector_base/README.rst:
```diff
@@ -0,0 +1,70 @@
+
+.
+
+Known issues / Roadmap
```
if empty remove it

elicoidal commented on odooconnector_base/README.rst:
```diff
@@ -0,0 +1,70 @@
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
```
https

elicoidal commented on odooconnector_base/__openerp__.py:
```diff
@@ -0,0 +1,32 @@
+{
+    'name': 'Odoo Connector - Base',
+    'summary': 'Base connector for Odoo2Odoo scenarios',
+    'version': '0.0.1',
```
version 8.0.1.0.0

elicoidal commented on odooconnector_base/models/odooconnector.py:
```diff
@@ -0,0 +1,169 @@
+    username = fields.Char(
+        string='Username',
+        required=True,
+        help='Username in the external Odoo Backend'
```
period

elicoidal commented on odooconnector_base/models/odooconnector.py:
```diff
@@ -0,0 +1,169 @@
+    password = fields.Char(
+        string='Password',
+        required=True,
+        help='Password'
```
no need

elicoidal commented on odooconnector_base/models/odooconnector.py:
```diff
@@ -0,0 +1,169 @@
+    export_backend_id = fields.Integer(
+        string='Backend ID in the external system',
+        help="""The backend id that represents this system in the external
+                system """
```
period

elicoidal commented on odooconnector_base/models/odooconnector.py:
```diff
@@ -0,0 +1,169 @@
+    export_partner_id = fields.Integer(
+        string='Partner ID in the external System',
+        help="""The partner id that represents this company in the
+                external system"""
```
period(s)
(several)

elicoidal commented on odooconnector_base/models/odooconnector.py:
```diff
@@ -0,0 +1,169 @@
+
+    default_export_backend = fields.Boolean(
+        string='Default Export Backend',
+        help='Use this backend as an automatic export targed'
```
s/targed/target

elicoidal commented on odooconnector_base/models/partner.py:
```diff
@@ -0,0 +1,120 @@
+              ('mobile', 'mobile'), ('fax', 'fax'), ('email', 'email'),
+              ('comment', 'comment'), ('supplier', 'supplier'),
+              ('customer', 'customer'), ('ref', 'ref'), ('lang', 'lang'),
+              ('date', 'date'), ('notify_email', 'notify_email')]
```
what about "active"

maljac replied:
I had some trouble with `active` in combination with `product.product` and `product.template`. Honestly: I haven't tried it in the partner object. Good point!

elicoidal commented on odooconnector_base/models/product.py:
```diff
@@ -0,0 +1,268 @@
+              ('purchase_ok', 'purchase_ok'), ('sale_ok', 'sale_ok'),
+              ('lst_price', 'lst_price'), ('standard_price', 'standard_price'),
+              ('ean13', 'ean13'), ('default_code', 'default_code'),
+              ('description', 'description')]
```
I don't know if it is a good idea to sync 
- type => most of the time different between companies
- purchase_ok => in icops, usually the opposite between companies
- sale_ok => in icops, usually the opposite between companies
- lst_price => those values are usually different between companies, all the more with currencies
- standard_price => those values are usually different between companies, all the more with currencies

maljac replied:
I understand your objections and in some scenarios they are true. In the scenarios that I implemented so far it was necessary to sync all those fields.

What I did in a private fork (which I will open source in Q1/16) is the configuration option (in the frontend!) to select fields that should be excluded from syncing.

What we could discuss is:
1. should we sync max fields and exclude fields that should not be synced or 
2. should we sync min viable number of fields and define additional fields per use case

_Pro / Cons (1)_

Pro: Some more complex logic than `field1=field1` can be used without dev

Con: "Exclude from Sync" list might be long if you need only min. viable fields

elicoidal commented on odooconnector_base/models/product.py:
```diff
@@ -0,0 +1,268 @@
+
+    direct = [('name', 'name'), ('name_template', 'name_template'),
+              ('type', 'type'),
+              ('purchase_ok', 'purchase_ok'), ('sale_ok', 'sale_ok'),
```
same remarks as above

elicoidal commented on odooconnector_base/unit/backend_adapter.py:
```diff
@@ -0,0 +1,147 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+
+import oerplib.error
```
Should not we protect the import?

maljac replied:
You mean like trying to import and catch a import error for the case that the external dependencie (oerplib) is not installed?

``` python

try:
    import oerplib.error
except ImportError:
    pass   
```

elicoidal replied:
yes.
You should probably log the error as well (there is probably some standard code in some OCA module for this kind of import)

elicoidal commented on odooconnector_po2so/__openerp__.py:
```diff
@@ -0,0 +1,28 @@
+{
+    'name': 'Odoo Connector - PO2SO',
+    'summary': 'PO2SO Process Addon',
+    'version': '0.0.1',
```
8.0.1.0.0

elicoidal commented on odooconnector_po2so/__openerp__.py:
```diff
@@ -0,0 +1,28 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    'name': 'Odoo Connector - PO2SO',
+    'summary': 'PO2SO Process Addon',
```
Purchase to SO Intercompany process addon.

elicoidal commented on odooconnector_purchase/README.rst:
```diff
@@ -0,0 +1,64 @@
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
```
https

elicoidal commented on odooconnector_purchase/__openerp__.py:
```diff
@@ -0,0 +1,26 @@
+{
+    'name': 'Odoo Connector - Purchase',
+    'summary': 'Technical base for purchase related Odoo Connector scenarios',
+    'version': '0.0.1',
```
8.0.1.0.0

Commit:
Fixes typos / Catch import error for external depency

maljac commented:
@elicoidal Thanks for the review. FYI: I'm planning to spend more time in Q1 to push the connector to a good state.

guewen commented on odooconnector_base/events.py:
```diff
@@ -0,0 +1,142 @@
+#           export jobs will be created: one for product.template and one for
+#           product.product. If you have more than one export backend, this
+#           gets even multiplied by the amount of backends: 2 backends means
+#           4 export jobs. So far I have no idea how to deal with this!
```
you could use a custom (or even better, add it in `connector_base_product`) event `on_product_write` handled specially:
- In the write of `product.product`, you pass their ids in a context key, that will be read by the template's write (https://github.com/OCA/connector-ecommerce/blob/9.0/connector_ecommerce/product.py#L128-L130)
- In the write of `product.template`, if you come from `product.product` with the context key, you don't re-trigger the event for those ids: https://github.com/OCA/connector-ecommerce/blob/9.0/connector_ecommerce/product.py#L67-L79

liebana commented:
Can somebody point out some functional information on this?

maljac commented:
@liebana : Not sure what you mean? Information about guewen's comment?

liebana commented:
Sorry, it was a generic question because I wasn't reading the updated readme.rst file commited. I was only checking what was available to connect two Odoo instances in two cases:
- Consolidate sales from multiple POS stores in a central one.
- Create a sale order based on a purchase order in the other instance.

It seems that this would cover both cases, based on your generic development.

Best regards.

yelizariev commented:
FYI: we test and migrate the `odooconnector_base` to 10.0: https://github.com/it-projects-llc/connector-odoo2odoo/tree/10.0-migration
(I don't think that it's time to make a PR, so I just share the source for now)
